File: climb-stairs/climbStairs.java
import java.util.*; 
import java.lang.*; 

public class climbStairs {
    public static void main(String[] args) {
        int y = 9;
        System.out.println(climbStairs(y));
    }

    public static int climbStairs(int n) {
        if (n == 0) return 0;
        if (n == 1) return 1;
        int case1 = 1;
        int case2 = 1;
        
        for(int i = 2; i <= n; i++){
            int temp = case2;
            case2 = temp+case1;
            case1 = temp;
        }
        return case2;
    }
}

File: GO/climb-stairs/climbStairs.go
package main

import "fmt"

func climbStairs(n int) int{
	if n == 0 {return 0}
	if n == 1 {return 1}
	case1 := 1
	case2 := 2

	for i := 2; i <n; i++ {
		case1, case2 = case2, case1+case2
	}
	return case2
}

func main()  {
	fmt.Println(climbStairs(9))
}